lib/engine/action/propose.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+
+module Engine
+  module Action
+    class Propose < Base
+      attr_reader :entity, :corporation, :company, :price
+
+      def initialize(entity, corporation:, company:, price:)
```

Comment:
do you need corporation? isn't entity the corporation?

Reply:
No, entity is the player.